.github/install_open_csp.sh

```diff
@@ -1,103 +1,206 @@
     echo "Moving to $MW_PATH"
     cd $MW_PATH || exit 1
 
-    if [ $SKIP_STEPS -le $CURRENT_STEP ]; then
-        echo ">>> Step $CURRENT_STEP: Setting up \"LocalSettings.php\""
-        setup_localsettings
-    fi
-    CURRENT_STEP=$(($CURRENT_STEP+1))
-
-    if [ $SKIP_STEPS -le $CURRENT_STEP ]; then
-        echo ">>> Step $CURRENT_STEP: Running composer."
-        do_composer
-    fi
-    CURRENT_STEP=$(($CURRENT_STEP+1))
-
-    if [ $SKIP_STEPS -le $CURRENT_STEP ]; then
-        echo ">>> Step $CURRENT_STEP: Running maintenance scripts"
-        run_maintenance_scripts
-    fi
-    CURRENT_STEP=$(($CURRENT_STEP+1))
-
-    if [ $SKIP_STEPS -le $CURRENT_STEP ]; then
-        echo ">>> Step $CURRENT_STEP: Running PageSync scripts"
-        run_pagesync_scripts
-    fi
-    CURRENT_STEP=$(($CURRENT_STEP+1))
-
-    if [ $SKIP_STEPS -le $CURRENT_STEP ]; then
-        echo ">>> Step $CURRENT_STEP: Running rebuildData"
-        run_rebuild_data
-    fi
-    CURRENT_STEP=$((CURRENT_STEP+1))
-
-    echo "Moving to $OLD_PATH"
-    cd $OLD_PATH
+    for step in "${STEPS_TO_RUN[@]}"; do
+        if [ x$AUTOMATIC == x ]; then # Prompt user if not in auto mode
+            read -p "Next step: '${MESSAGES[$step]}' Continue? [Y/n] " -n 1 input
+            if [[ $input =~ ^[Nn]$ ]]; then
+                echo -e "\nAborting installation."
+                exit 1
+            fi
+        fi
+        echo -e "\n${TSC[bold]}${TSC[green]}${MESSAGES[$step]} ...${TSC[reset]}"
+        ${STEPS[$step]} || exit_with_message # Do the thing
+        # If $step is nonzero, write it to the tempfile.
+        if [ $step -gt 0 ]; then
+            echo -n $step > $TEMPFILE_PROGRESS
+        fi
+    done
 
     succes_message
 }
 
-usage()
-{
-    echo "Usage: $0 <mediawiki path> [<steps to skip>]";
-    echo "Example: $0 \"/var/www/wiki/\""
-    echo ""
-    echo "<mediawiki path>:   The full path to your mediawiki installation"
-    echo "<steps to skip>:    An integer that says how many steps to skip: 0 is full installation, 1 does all but the first step, etc."
-}
-
-exit_with_message()
+function exit_with_message()
 {
-    echo "\n\nThe installation was unsuccessfull."
+    echo -e "\n\nThe installation was unsuccessfull."
     echo "Please read the error messages above, and act accordingly."
     echo ""
     echo "It could be that your problem is already mentioned on our Installation troubleshooting:"
     echo "https://github.com/Open-CSP/open-csp/blob/$CSP_BRANCH/.github/INSTALLATION_FAQ.md"
     echo ""
-    echo "After resolving the issue, you can continue installation from the current step with the following command:"
-    echo "$0 \"$MW_PATH\" $CURRENT_STEP"
+    echo "After resolving the issue, you may resume using the following command:"
+    echo "$CMD"
 
     exit 1
 }
 
-validate_mw_path()
+function validate_mw_path()
 {
+    return 0
     if [ -d "$1" ]; then
         if [ -e "$1/composer.json" ]; then
             # Use php for checking the json.
```

Comment:
Shouldn't this php code also be done on the altpath? The `[ $? -eq 0 ]` will give `false` when the provided path does not point to a mediawiki installation.

Reply:
Wait, what? But the only purpose of altpath is to find if someone meant to put a subdirectory of what they actually put, limiting the search to a depth of 4 to prevent an everlasting search if someone for some weird reason puts their mediawiki path as "/" or whatever.  So by definition the escalation of the originally provided path into finding the altpath is an action you do not want to repeat that on the altpath or you're creating a loop.

The [ $? -eq 0 ] I didn't really look at I think. That's part of the code I left alone and didn't revise or even review at all. To me it seems like it would match if the php command thing before has finished executing correctly, no? Because 0 may be false, but in exit codes 0 is success?